eth2/operation_pool/src/attestation.rs

```diff
@@ -1,35 +1,53 @@
 use crate::max_cover::MaxCover;
-use types::{Attestation, BeaconState, BitList, EthSpec};
+use state_processing::common::{get_attesting_indices, get_base_reward};
+use std::collections::HashMap;
+use std::convert::TryInto;
+use types::{Attestation, BeaconState, BitList, ChainSpec, EthSpec};
 
 pub struct AttMaxCover<'a, T: EthSpec> {
     /// Underlying attestation.
     att: &'a Attestation<T>,
-    /// Bitfield of validators that are covered by this attestation.
-    fresh_validators: BitList<T::MaxValidatorsPerCommittee>,
+    /// Mapping of validator indices and their rewards.
+    fresh_validators_rewards: HashMap<u64, u64>,
 }
 
 impl<'a, T: EthSpec> AttMaxCover<'a, T> {
     pub fn new(
         att: &'a Attestation<T>,
-        fresh_validators: BitList<T::MaxValidatorsPerCommittee>,
+        state: &BeaconState<T>,
+        spec: &ChainSpec,
+        total_active_balance: u64,
     ) -> Self {
+        let fresh_validators = earliest_attestation_validators(att, state);
+        let indices = get_attesting_indices(state, &att.data, &fresh_validators)
+            .expect("should have returned valid indices");
+        let fresh_validators_rewards: HashMap<u64, u64> = indices
+            .iter()
+            .cloned()
+            .map(|validator_index| validator_index as u64)
+            .zip(indices.iter().cloned().map(|validator_index| {
+                get_base_reward(state, validator_index, total_active_balance, spec)
+                    .expect("should have returned base reward for validator")
+                    / spec.proposer_reward_quotient
+            }))
+            .collect();
```

Comment:
I think this could be written more clearly and cleanly without the `zip`, perhaps something like:

```rust
indices
    .iter()
    .map(|i| *i as u64)
    .flat_map(|validator_index| {
        let reward =
            get_base_reward(state, validator_index, total_active_balance, spec).ok()? /
                spec.proposer_reward_quotient;
        Some((validator_index, reward))
    })
    .collect()
```

It'll silently drop validators that error in `get_base_reward` (should be none of them)